### PR TITLE
refactor(components): [tooltip] switch to script-setup syntax

### PR DESCRIPTION
**Pull request by LinZhanMing**
#6114 
Please make sure these boxes are checked before submitting your PR, thank you!

- [x] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [x] Make sure you are merging your commits to `dev` branch.
- [x] Add some descriptions and refer to relative issues for your PR.

### CODE REVIEW

File: packages/components/tooltip/src/tooltip.vue [hunk 1]
```diff
@@ -48,30 +48,20 @@
 </template>
 
 <script lang="ts">
```
[the remark this answers is not in the document]
Comment (LinZhanMing) [reply packages/components/tooltip/src/tooltip.vue:50]: i tried, but it reported an error, and say

```
`defineProps()` in <script setup> cannot reference locally declared variables
use a separate normal <script> to export the options instead.
```

image

[the remark this answers is not in the document]
Comment (sxzz) [reply packages/components/tooltip/src/tooltip.vue:50]: `defineOptions` for `name` property, and use `defineProps/emits` for props and emits.

**Comment by holazz** [comment]
Hi! just passing by here when I browsed PRs related to the refactor plan. I have refactored some components, and want to share some experience with you here, hope it will be useful to you.

- rename `index.vue` to `[component-name].vue`

- extract `[component-name].vue` props and emits to `[component-name].ts`

  -	  refactor
	  ```vue
	  // packages/components/[component-name]/[component-name].vue
	  
	  <script lang="ts">
	  import { defineComponent } from 'vue'
	  export default defineComponent({
	    name: El[ComponentName],
	    components: {
	      ...
	    },
	    inheritAttrs: xxx,
	    props: {
		  ...
	    },
	    emits: [...],
	    setup(props) {
	      return {
	        ...
	      }
	    },
	  })
	  </script>
	  ```
	  
	  to
	  
	  ```vue
	  // packages/components/[component-name]/[component-name].vue
	  
	  <script lang="ts" setup>
	  import { [componentName]Emits, [componentName]Props } from './[component-name].ts'
	  
	  defineOptions({
	    name: El[ComponentName],
	    inheritAttrs: xxx,
	  })
	  
	  const props = defineProps([componentName]Props)
	  const emit = defineEmits([componentName]Emits)
	  
	  ...
	  
	  // if needed
	  defineExpose({
	    // only expose what user needed
	    ...
	  })
	  </script>
	  ```

	  ```ts
	  // packages/components/[component-name]/[component-name].ts
	  
	  import {
	    buildProps,
	    definePropType,
	  } from '@element-plus/utils'
	  
	  import type { ExtractPropTypes } from 'vue'
	  import type [ComponentName] from './[component-name].vue'
	  
	  export const [componentName]Props = buildProps({
	    // refactor `xxx as PropType` to `definePropType`
	  
	    /**
	    refactor
	    
	    size: {
	      type: definePropType<ComponentSize>(String),
	      values: componentSizes,
	    },
	    
	    to
	    
	    size: useSizeProp,
	    */
	    ...
	  } as const)
	  export type [ComponentName]Props = ExtractPropTypes<typeof [componentName]Props>
	  
	  export const [componentName]Emits = {
	    ...
	  }
	  export type [ComponentName]Emits = typeof [componentName]Emits
	  
	  export type [ComponentName]Instance = InstanceType<typeof [ComponentName]>
	  
	  ```

- extract tokens(if exists)
	- move `packages/components/[component-name]/tokens.ts` to `packages/tokens/[component-name].ts`, and export it from `packages/tokens/index.ts`

- extract composables functions(if exists)
	- extract `useXXX` from `.vue`

- use `withInstall`

refactor

```ts
// packages/components/[component-name]/index.ts

import [ComponentName] from './src/[component-name].vue'

import type { App } from 'vue'
import type { SFCWithInstall } from '@element-plus/utils'

[ComponentName].install = (app: App): void => {
  app.component([ComponentName].name, [ComponentName])
}

const _[ComponentName] = [ComponentName] as SFCWithInstall<typeof [ComponentName]>

export default _[ComponentName]
export const El[ComponentName] = _[ComponentName]

export * from './src/[component-name]'
```

to

```ts
// packages/components/[component-name]/index.ts

import { withInstall } from '@element-plus/utils'

import [ComponentName] from './src/[component-name].vue'

export const [ComponentName] = withInstall([ComponentName])
export default [ComponentName]

export * from './src/[component-name]'
```

- use utils (`@element-plus/utils`)
  - `typeof xxx === undefined` -> `isUndefined(xxx)`(`isArray`, `isNumber`...)
  - `xxx.length === 0` -> `isEmpty(xxx)`
  - `isString(xxx) ? xxx ? ${xxx}px` -> `addUnit(xxx)`
  - ...

**Comment by xiaoxian521** [comment]
Hey, thank you very much for your contribution, but here are some conflicts that need to be solved by you. We can continue to review the code after solving. One thing to remind is that if the conflicts is still not resolved after `10` days, sorry, we will close this pr ❤️

**Pull request closed without merging**